Cleanup for the Stash remote implementation

josmo opened the pull request:
Started to move the implementation closer to that of Bitbucket cloud
Still need to refactor out parsing and conversion

josmo commented:
@bradrydzewski I wanted to see if you had a sec to look at this and suggestions?  I tried to have it follow suit on the bitbucket cloud remote and I'll start getting better test coverage next.

bradrydzewski commented:
@josmo sorry for the lack of communication on my end. This is looking great. Did you want to add unit tests as part of this PR or a follow-up? Shoot me a note in the drone-dev room and we can figure out a plan

josmo commented:
@bradrydzewski no worries at all I imagine you're pretty busy :)  I'll hit you up on the drone-dev room. I was planning on doing a different PR for the tests.

bradrydzewski commented on remote/bitbucketserver/bitbucketserver.go:
```diff
@@ -4,241 +4,150 @@ package bitbucketserver
-	"net/http"
-	"net/url"
-
 	log "github.com/Sirupsen/logrus"
```
in general we should try to avoid logging at such low levels. Logging is generally at a higher level, for example, in the remote, or when possible, in the handlers. Let's remove logging from this tier

bradrydzewski commented on remote/bitbucketserver/internal/client.go:
```diff
@@ -0,0 +1,234 @@
+	accessToken string
+}
+
+func NewClientWithToken(url string, Consumer *oauth.Consumer, AccessToken string) *Client {
```
variable names should be lower case (ie `consume *oauth.Consumer`)

bradrydzewski commented on remote/bitbucketserver/internal/client.go:
```diff
@@ -0,0 +1,234 @@
+		return nil, err
+	}
+
+	ModelUser := &model.User{
```
lower case variable names

bradrydzewski replied:
also if you look at the bitbucket implementation, the bitbucket client returns native bitbucket types. These types are then converted in a `convert.go` file. This allows use to change the drone model without having to change the client. Also much easier to unit test.
https://github.com/drone/drone/blob/master/remote/bitbucket/convert.go

bradrydzewski commented on remote/bitbucketserver/internal/client.go:
```diff
@@ -0,0 +1,234 @@
+		return nil, err
+	}
+	repo := &model.Repo{
+		Name:      bsRepo.Slug,
```
return a native bitbucket type and move the conversion to a `convert.go` file in the parent package

bradrydzewski commented on remote/bitbucketserver/internal/client.go:
```diff
@@ -0,0 +1,234 @@
+	var repos = []*model.RepoLite{}
+	for _, repo := range repoResponse.Values {
+		repos = append(repos, &model.RepoLite{
+			Name:     repo.Slug,
```
return a native bitbucket type and move the conversion to a `convert.go` file in the parent package

bradrydzewski commented on remote/bitbucketserver/internal/types.go:
```diff
@@ -0,0 +1,102 @@
+}
+
+type BSRepo struct {
+	Forkable bool `json:"forkable"`
```
maybe call this just `Repo`?

bradrydzewski commented on remote/bitbucketserver/types.go:
```diff
@@ -84,89 +84,4 @@ type postHook struct {
 	} `json:"repository"`
 }
```
this file should be removed all together and the structs in it could be moved to `internal/types.go`

josmo commented:
@bradrydzewski I think I addressed everything you had. Let me know if you see anything else.  I also fixed a couple minor things like the messages not being included on builds which should make it nicer for folks.  I'll have a few more PRs coming soon for adding tests, trying to figure out a clean PR build option, etc

appleboy commented:
@josmo Please help conflicts.

bradrydzewski commented:
@josmo can you reserve conflict and we can get this merged? :)

appleboy commented on remote/bitbucketserver/convert.go:
```diff
@@ -0,0 +1,110 @@
+
+	build := &model.Build{
+		Commit: hook.RefChanges[0].ToHash, // TODO check for index value
+		//Link: TODO find link
```
remove this comment. Link has been include as below.

josmo replied:
@appleboy I'll remove it in the next PR :)

appleboy commented:
@josmo Nice Improvement.

@bradrydzewski Can we merge this PR?

bradrydzewski commented:
absolutely, merging now

josmo commented:
@bradrydzewski Thanks! :)
